src/System.Runtime/tests/System/StringTests.cs

```diff
@@ -23,6 +23,7 @@ public static class StringTests
         [InlineData(new char[] { 'a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', '\0', 'i', 'j' }, "abcdefgh")]
         [InlineData(new char[] { 'a', '\0' }, "a")]
         [InlineData(new char[] { '\0' }, "")]
+        [InlineData(new char[] { '?', '@', ' ', '\0' }, "?@ ")] // ? and @ don't have overlapping bits
```

Comment:
The current implementation processes 2 chars at a time by checking if `(s[0] & s[1]) != 0`, namely if they have 1 bit in common. However this doesn't necessarily imply either of the chars are 0, e.g. `@` is 0b1000000 and `?` is 0b0111111 and `&`-ing them will give 0. So that makes us fall back to the slow path where we check individually each char for 0.